utilities/src/conversion.rs

```diff
@@ -95,59 +93,94 @@ where
     F::from_le_bytes_mod_order(output)
 }
 
-/// Invertible, deterministic, infallible conversion from arbitrary bytes to
+/// Deterministic, infallible, invertible conversion from arbitrary bytes to
 /// field elements.
 pub fn bytes_to_field_elements<B, F>(bytes: B) -> Vec<F>
 where
     B: AsRef<[u8]>,
     F: Field,
 {
-    // Need to ensure that F::characteristic is large enough to hold a u64.
+    // Ensure that F::characteristic is large enough to hold a u64.
     // This should be possible at compile time but I don't know how.
     // Example: could use <https://docs.rs/static_assertions> but then you hit
     // <https://users.rust-lang.org/t/error-e0401-cant-use-generic-parameters-from-outer-function/84512>
-    assert!(F::BasePrimeField::MODULUS_BIT_SIZE > 64);
+    assert!(
+        F::BasePrimeField::MODULUS_BIT_SIZE > 64,
+        "base prime field modulus bit len {} should exceed 64",
+        F::BasePrimeField::MODULUS_BIT_SIZE
+    );
+
+    if bytes.as_ref().is_empty() {
+        return Vec::new();
+    }
+
+    // various quantities
+    let primefield_bytes_len = usize::try_from((F::BasePrimeField::MODULUS_BIT_SIZE - 1) / 8)
+        .expect("prime field modulus byte len should fit into usize");
+    let extension_degree =
+        usize::try_from(F::extension_degree()).expect("extension degree should fit into usize");
```

Comment:
these checks seem a little redundant.
I guess "infallible" really means a success-or-nothing guarantee, this code is panic-able anyway, `usize.try_from().expect()` is too verbose than just `xx as usize` which is the same effect, correct?

Reply:
`xx as usize` is not the same effect.
[Operator expressions - The Rust Reference](https://doc.rust-lang.org/reference/expressions/operator-expr.html#semantics):
> Casting from a larger integer to a smaller integer (e.g. u32 -> u8) will truncate

We want it to panic in the event of overflow.